WordPress/Classes/ViewRelated/Menus/Views/MenuItemCheckButtonView.m

```diff
@@ -0,0 +1,158 @@
+#import "MenuItemCheckButtonView.h"
+#import "WPStyleGuide.h"
+#import "WPFontManager.h"
+#import "Menu+ViewDesign.h"
+
+@import Gridicons;
+
+static CGFloat const iconPadding = 3.0;
+
+@interface MenuItemCheckButtonView ()
+
+@property (nonatomic, strong) UIImageView *iconView;
+@property (nonatomic, assign) BOOL drawsHighlighted;
+@property (nonatomic, assign) CGPoint touchesBeganLocation;
+
+@end
+
+@implementation MenuItemCheckButtonView
+
+- (id)init
+{
+    self = [super init];
+    if (self) {
+     
+        self.translatesAutoresizingMaskIntoConstraints = NO;
+        self.backgroundColor = [UIColor whiteColor];
+        self.contentMode = UIViewContentModeRedraw;
+        
+        [self initIconView];
+        [self initLabel];
+    }
+    
+    return self;
+}
+
+- (void)initIconView
+{
+    UIImageView *iconView = [[UIImageView alloc] init];
+    iconView.translatesAutoresizingMaskIntoConstraints = NO;
+    iconView.image = [Gridicon iconOfType:GridiconTypeCheckmark];
+    iconView.tintColor = [WPStyleGuide mediumBlue];
+    iconView.contentMode = UIViewContentModeScaleAspectFit;
+    iconView.alpha = 0.0;
+    [self addSubview:iconView];
+    
+    [NSLayoutConstraint activateConstraints:@[
+                                              [iconView.topAnchor constraintEqualToAnchor:self.topAnchor constant:iconPadding],
+                                              [iconView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor constant:iconPadding],
+                                              [iconView.widthAnchor constraintEqualToAnchor:iconView.heightAnchor],
+                                              [iconView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor constant:-iconPadding]
+                                              ]];
+    self.iconView = iconView;
```

Comment:
For properties owned by the current class, it's recommended to not use `self.` during initialization.

Check out section "Don’t Use Accessor Methods in Initializer Methods and dealloc" in [this link](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmPractical.html).